src/java.base/windows/native/libnio/ch/FileDispatcherImpl.c

```diff
@@ -392,6 +394,57 @@ Java_sun_nio_ch_FileDispatcherImpl_size0(JNIEnv *env, jobject this, jobject fdo)
     return (jlong)size.QuadPart;
 }
 
+JNIEXPORT jint JNICALL
+Java_sun_nio_ch_FileDispatcherImpl_available0(JNIEnv *env, jobject this, jobject fdo)
+{
+    HANDLE handle = (HANDLE)(handleval(env, fdo));
+    jlong available;
+    if (handleAvailable((jlong)handle, &available)) {
```

Comment:
Ugh, we shouldn't be using handleAvailable here. This native method needs to stand on its own.

Reply:
When/if we get to the point that FIS/FOS/RAF is implemented on FileChannel then it will become clearer.